guibot/desktopcontrol.py

```diff
@@ -728,10 +728,10 @@ def capture_screen(self, *args):
 
         See base method for details.
         """
-        xpos, ypos, width, height, filename = self._region_from_args(*args)
-        self._backend_obj.captureRegion(filename, xpos, ypos, width, height)
-        pil_image = PIL.Image.open(filename).convert('RGB')
-        os.unlink(filename)
+        xpos, ypos, width, height, _ = self._region_from_args(*args)
+        self._backend_obj.refreshScreen()
+        cropped = self._backend_obj.screen.crop((xpos, ypos, xpos + width, ypos + height))
+        pil_image = cropped.convert('RGB')
```

Comment:
This looks great, I guess a greater familiarity with the contemporary VNCDoTool API definitely helps!